cache.go

```diff
@@ -67,7 +75,21 @@ func (c *cache) Set(k string, x interface{}, d time.Duration) {
 	c.mu.Unlock()
 }
 
+// Check to see if this element is going to be replaced or is it going to be
+// added. If it's added, then we can't exceed the maxElements set by the user.
+func (c *cache) canSet(k string) bool {
+	if c.maxElements > 0 && len(c.items) >= c.maxElements {
```

Comment:
Since `Set()` and therefor canSet is protected by a WriteMutex, why not hold a counter in memory, instead of performing len() on the map?